src/state/nfts/createNftsStore.ts

```diff
@@ -438,88 +439,87 @@ export const createNftsStore = (address: Address | string) =>
-              hasCollection: !!collection,
-              isEmpty: collection ? isEmpty(collection) : 'N/A',
-            });
-            if (collection && !isEmpty(collection)) {
-              logger.debug(`🔍 [NFT Store] Collection exists and not empty, returning existing nftsByCollection`);
-              return Promise.resolve(nftsByCollection);
-            }
+          if (!needsFetch) {
+            logger.debug('🔍 [NFT Store] No fetch needed for showcase/hidden, returning existing nftsByCollection');
+            return;
+          }
+        } else {
+          const collection = getNftsByCollection(normalizedCollectionId);
+          logger.debug(`🔍 [NFT Store] Checking regular collection: ${normalizedCollectionId}`, {
+            normalizedCollectionId,
+            hasCollection: !!collection,
+            isEmpty: collection ? isEmpty(collection) : 'N/A',
+          });
+          if (collection && !isEmpty(collection)) {
+            logger.debug(`🔍 [NFT Store] Collection exists and not empty, returning existing nftsByCollection`);
+            return;
           }
         }
 
-        const promise = fetch(
-          { collectionId: normalizedCollectionId, isMigration },
-          { skipStoreUpdates: true, updateQueryKey: false, cacheTime: time.infinity, staleTime: time.infinity }
-        )
-          .then(data => {
-            if (!data) return nftsByCollection;
-
-            const now = Date.now();
-
-            const newNftsByCollection = new Map([...nftsByCollection, ...data.nftsByCollection]);
-            set(state => ({
-              nftsByCollection: newNftsByCollection,
-              fetchedCollections: { ...state.fetchedCollections, [normalizedCollectionId]: now },
-            }));
-            return newNftsByCollection;
-          })
-          .catch(error => {
-            logger.error(new RainbowError(`Failed to fetch NFT collection: ${normalizedCollectionId}`, error));
-            return nftsByCollection;
-          })
-          .finally(() => {
-            // Clean up the promise when it's done
-            const addressPromises = collectionsPromises.get(address);
-            if (addressPromises) {
-              addressPromises.delete(normalizedCollectionId);
-              // Clean up the address entry if no more promises
-              if (addressPromises.size === 0) {
-                collectionsPromises.delete(address);
+        const promise = (async () => {
+          for (let attempt = 1; attempt <= MAX_RETRIES; attempt++) {
+            try {
+              const data = await fetch(
+                { collectionId: normalizedCollectionId },
+                { skipStoreUpdates: true, cacheTime: time.infinity, staleTime: time.infinity }
+              );
+
+              if (!data) continue;
+
+              const now = Date.now();
+              set(state => ({
+                nftsByCollection: new Map([...state.nftsByCollection, ...data.nftsByCollection]),
```

Comment:
actual fix is here! Rest of the changes are improvements as well / moving out business logic that doesn't need to exist in this function call.